drops/DropsDriver.py

```diff
@@ -67,21 +79,35 @@ def __del__(self):
   Here we define specific interactions that the machine is capable of fielding
   These actions are exposed as member functions of this driver class. 
   '''
-
   def middle_invocation_wrapper(func):
     '''
     Define a decorator function to adorn all of these 'high' middleware calls
     Logs what functions is being called and returns the response. 
     '''
-    def inner(self, *args):
+    @wraps(func) # should pass func name
+    def wrapped_func_that_gets_and_returns_response(self, *args):
       logger.info(f"Invoking {func.__name__}")
       func(self, *args)
       resp = self.get_response()
+
+      if self.should_api_calls_block == True:
+        logging.info("Calling with blocking set to true")
+        status = resp.STATUS['Status'] 
+        while (status != 'Idle'):
+          logging.info(f"Status is: {status}")
+          time.sleep(.5)
+          status = self.get_response()
```

Comment:
Did you mean to use self.get_status? If the status is not being updated, the loop will block indefinitely.